### PR TITLE
Remove omnibus Berksfile and Gemfile locks and update bundler/rubygems

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
@jmink @manderson26 @chef/engineering-services

**Comment by schisamo** [comment]
:+1:

**Comment by scotthain** [comment]
:+1:

**Comment by schisamo** [comment]
Let's hold off merging this until I can deploy https://github.com/chef-cookbooks/opscode-ci/pull/412 (Mac OS X in the push-jobs-client pipeline).

File: omnibus/config/projects/push-jobs-client.rb [hunk 1]
```diff
@@ -35,12 +35,14 @@
   install_dir "#{default_root}/#{name}"
 end
 
-override :bundler,        version: "1.7.12"
 # Uncomment to pin the chef version
 #override :chef,           version: "12.2.1"
```
Comment (ksubrama) [inline omnibus/config/projects/push-jobs-client.rb]: If you're going to use this to make a release, remember to pin chef.  You don't want to release some random untested build from master.

Comment (ksubrama) [inline omnibus/config/projects/push-jobs-client.rb]: Also - chef (our software definition) pulls in ohai which also picks up master by default.  Pin it to whatever chef pinned it to when it was released.

**Pull request merged**